src/utils/hsdis/Makefile

```diff
@@ -159,16 +175,45 @@ BINUTILSDIR	= $(shell cd $(BINUTILS);pwd)
 endif
 
 CPPFLAGS	+= -I$(BINUTILSDIR)/include -I$(BINUTILSDIR)/bfd -I$(TARGET_DIR)/bfd
-CPPFLAGS	+= -DLIBARCH_$(LIBARCH) -DLIBARCH=\"$(LIBARCH)\" -DLIB_EXT=\"$(LIB_EXT)\"
+else
+# LLVM
+LLVMDIR		= $(shell cd $(LLVM);pwd)
+
+CFLAGS		+= -DLLVM=1
+CPPFLAGS	+= -I$(LLVMDIR)/include
+endif
+
+CPPFLAGS	+= -DLIBOS_$(OS) -DLIBOS=\"$(OS)\" -DLIBARCH_$(LIBARCH) -DLIBARCH=\"$(LIBARCH)\" -DLIB_EXT=\"$(LIB_EXT)\"
 
 TARGET_DIR	= build/$(OS)-$(JDKARCH)
 TARGET		= $(TARGET_DIR)/hsdis-$(LIBARCH)$(LIB_EXT)
 
-SOURCE		= hsdis.c
+SOURCE		= hsdis.cpp
 
+ifeq ($(LLVM),)
 LIBRARIES =	$(TARGET_DIR)/bfd/libbfd.a \
 		$(TARGET_DIR)/opcodes/libopcodes.a \
 		$(TARGET_DIR)/libiberty/libiberty.a
+else
+LIBRARIES/amd64 = LLVMX86Disassembler LLVMX86AsmParser LLVMX86CodeGen LLVMCFGuard LLVMGlobalISel LLVMSelectionDAG \
```

Comment:
To disassemble code, I don't think we have to link so many libraries.  It looks like code only explicitly depends LLVMMCDisassembler and LLVMTarget here. 

If we do need to link those libraries,  how about we just use `llvm-config --libs`. If we declare so many names here,  the Makefile is subject to LLVM.  In history, LLVM refactored a lot.

Reply:
>To disassemble code, I don't think we have to link so many libraries. It looks like code only explicitly depends LLVMMCDisassembler and LLVMTarget here.

This is the result of `llvm-config --libs x86 x86disassembler` and `llvm-config --libs aarch64 aarch64disassembler`.

> If we do need to link those libraries, how about we just use llvm-config --libs. If we declare so many names here, the Makefile is subject to LLVM. In history, LLVM refactored a lot.

That should work in the general case. I did it this way originally because of cases where we need to use a cross-compiled LLVM. Then `llvm-config` and the related libraries would be compiled for the host platform. Then, the user can just specify the `LIBRARIES/*` variable by hand on the command line.

Reply:
I saw you manged to solve this cross-compilation issue. cool!
Please note that I am not a reviewer. feel free to ignore what I said if it makes nonsense. 
I just want to chip in some constructive ideas.

Reply:
That isn't going to work in the case where you've compiled LLVM for a different host than the current one (build != host). I'll add a comment to document such a case and what to do then. Thanks for the reminder.

> Please note that I am not a reviewer. feel free to ignore what I said if it makes nonsense.
> I just want to chip in some constructive ideas.

This is more than welcome! Having another pair of eyes is always welcome, especially in the open from someone I don't have the opportunity to work with on a daily basis.